Accept JWT in the body or in parameters according to the method (#75656)
metabase/metabase: docs/people-and-groups/authenticating-with-jwt.md

5. In the event of a successful sign-in, your authentication app should issue a GET request to your Metabase endpoint with the token and the "return to" URI: `http://localhost:3000/auth/sso?jwt=TOKEN_GOES_HERE&return_to=/question/1-superb-question`.
6. Metabase verifies the JSON Web Token, logs the person in, then redirects the person to their original destination, `/question/1-superb-question`.

> As an alternative to putting the JWT in the URL query string (which some environments log or cache), you can send the same token in a `POST` request to `/auth/sso` with `Content-Type: application/json`. You can still pass `return_to` as a query parameter. The login behavior is otherwise the same as the GET flow.

Reviewer: Does this edit clarify the situation?

```suggestion
- For full app embeds, use a `GET` request.
- For modular embeds use a `POST` request with a JSON body (to avoid putting the JWT in the URL). You'll also want to use `POST` requests in contexts that log or cache URLs, such as server-side integrations or testing.

In both cases the login behavior is the same, and you can pass `return_to` as a query parameter in both `GET` and `POST` requests.
```

> For example, with curl:

Reviewer: ```suggestion
An example `POST` request with curl:
```

```bash
curl -X POST "http://localhost:3000/auth/sso?return_to=/question/1-superb-question" \
  -H "Content-Type: application/json" \
  -d '{"jwt": "TOKEN_GOES_HERE"}'
```

> From JavaScript (for example, modular embedding or programmatic login):

Reviewer: ```suggestion
Or with JavaScript:
```

```js
await fetch(`${METABASE_URL}/auth/sso?return_to=/question/1-superb-question`, {
  method: "POST",
  headers: { "Content-Type": "application/json" },
  body: JSON.stringify({ jwt: token }),
  credentials: "include",
});
```

> GET redirect remains the standard full-app browser flow. POST with a JSON body is for environments that want to avoid putting the JWT in the URL, such as the embedding SDK, server-side integrations, or testing.

Reviewer: Contigent on accepting the suggestion above.

```suggestion
```
